frontend/src/app/oferta/listagem-page-oferta/listagem-page-oferta.component.ts

```diff
@@ -0,0 +1,180 @@
+        private fb: FormBuilder,
+        private notification: PageNotificationService
+    ) { }
+
+    ngOnInit(): void{
+        this.iniciarForm();
+        this.buscarPorSituacao(true);
+        this.buscarCategorias();
+    }
+
+    iniciarForm () {
+        this.form = this.fb.group({
+          id: [null],
+          itemDtoId: [null,[Validators.required]],
+          usuarioDtoId: [null,[Validators.required]],
+          situacaoDtoId: [null,[Validators.required]],
+          itensOfertados: [null,[Validators.required]]
+        });
+    }
+
+    buscarCategorias() {
+        this.itemService.buscarCategorias().subscribe(
+            (categorias) => {
+                this.sortCategoria = categorias.map(categoria => {
+                    return {label: categoria.descricao, value: categoria.id}
+                });
+
+                this.sortCategoria.unshift({label: 'Pesquisar por Categoria', value: null});
+            }
+        )
+    }
+
+    buscarPorSituacao(situacao: boolean) {
+        this.itemService.buscarPorSituacao(situacao).subscribe(
+            (itens) => {
+                this.itens = this.removerItensUsuarioLogado(itens);
+                this.montaImagem(this.itens);
+            }
+        )
+    }
+
+    removerItensUsuarioLogado(itens: ItemModel[]){
+        let itensUsuario = itens.filter(item => {
+            return item.usuarioDtoId != this.localstorageService.getId();
+        })
+
+        return itensUsuario;
+    }
+
+    mostrarItem(item) {
+        this.selectedItem = item;
+        this.displayDetalhesItem = true;
+    }
+
+    realizaTroca(item){
+        this.displayTroca = true;
+        this.selectedItensOfertados = [];
+        this.idItemOferta = item.id;
+
+        this.itemService.buscarPorUsuario(this.localstorageService.getId())
+        .subscribe(
+            (itens) => {
+                this.availableItens = itens.filter(item => {
+                    return item.disponibilidade;
+                });
+                this.montaImagem(this.availableItens);
+            }
+        )
+    }
+
+    montaImagem(itens){
+        itens.forEach(item => {
+            let montandoBase64 = 'data:image/png;base64,';
+            let novaString = montandoBase64.concat(item.foto);
+            item.foto = novaString;
+        });
+    }
+
+    dragStart(event, item) {
+        this.draggedItem = item;
+    }
+
+    drop(event) {
+        if (this.draggedItem) {
+            let draggedItemIndex = this.findIndex(this.draggedItem, this.availableItens);
+            this.selectedItensOfertados = [...this.selectedItensOfertados, this.draggedItem];
+            this.availableItens = this.availableItens.filter((val,i) => i!=draggedItemIndex);
+            this.draggedItem = null;
+        }
+    }
+
+    dragEnd(event) {
+        this.draggedItem = null;
+    }
+
+    findIndex(item, listaItens) {
+        let index = -1;
+        for(let i = 0; i < listaItens.length; i++) {
```

Comment:
Não usar um for, se for um filtro pode usar um filter, e for pra achar um cara em específico usa um find